src/components/Events/EventsTable.js

```diff
@@ -0,0 +1,185 @@
+
+import { SEVERITY, DATE, MESSAGE, EVENT_SEVERITY, UNKNOWN, filterEvents } from './EventFilters'
+
+import { SortFields } from './EventSort'
+
+const sortEvents = (events = [], { id, isAsc } = {}, locale) => {
+  if (!id) {
+    return
+  }
+  const direction = isAsc ? 1 : -1
+  const getField = (event, id) => {
+    switch (id) {
+      case SEVERITY:
+        return '' + EVENT_SEVERITY[event?.severity ?? UNKNOWN] ?? EVENT_SEVERITY[UNKNOWN]
+      case DATE:
+        return '' + event?.time ?? ''
+      case MESSAGE:
+        return event?.description
+    }
+  }
+
+  events.sort((a, b) => direction * localeCompare(getField(a, id), getField(b, id), locale))
+}
+
+const EventsTable = ({
+  msg,
+  locale,
+  events,
+  eventFilters: { [SEVERITY]: severityFilters, [DATE]: dateFilters, [MESSAGE]: messageFilters },
+  eventSort = { id: DATE },
+  clearAllFilters,
+  setSort,
+}) => {
+  const columns = [
+    SortFields[SEVERITY],
+    SortFields[DATE],
+    SortFields[MESSAGE],
+  ].map(({ messageDescriptor, ...rest }) => ({
+    ...rest,
+    messageDescriptor,
+    label: messageDescriptor?.id ? translate({ id: messageDescriptor?.id, msg }) : '',
+  }))
+
+  const activeSortIndex = columns.findIndex(({ id }) => id === eventSort.id)
+  const activeSortDirection = eventSort.isAsc ? 'asc' : 'desc'
+  const buildSort = (columnIndex) => ({
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+    },
+    onSort: (_event, index, direction) => {
+      setSort({
+        isAsc: direction === 'asc',
+        ...SortFields?.[columns[index]?.id ?? SEVERITY],
+      })
+    },
+    columnIndex,
+  })
+
+  const filteredEvents = filterEvents({ events, severityFilters, dateFilters, messageFilters })
+
+  sortEvents(filteredEvents, eventSort, locale)
+
+  return (
+    <div className={style.container}>
+      { !filteredEvents && (
+        <EmptyState variant="xl" isFullHeight>
+          <EmptyStateIcon variant="container" component={Spinner} />
+        </EmptyState>
+      ) }
+
+      { filteredEvents?.length === 0 && (
+        <EmptyState variant="xl" isFullHeight>
+          <EmptyStateIcon icon={SearchIcon} />
+          <Title size="lg" headingLevel="h4">
+            {msg.noEventsFound()}
+          </Title>
+          <EmptyStateBody>{msg.clearAllFiltersAndTryAgain()}</EmptyStateBody>
+          <Button variant="link" onClick={clearAllFilters}>{msg.clearAllFilters()}</Button>
+        </EmptyState>
+      ) }
+
+      { filteredEvents?.length > 0 && (
+        <>
+          <TableComposable
+            aria-label={msg.events()}
+            variant='compact'
+            isStriped
+            isStickyHeader
+          >
+            <Thead>
+              <Tr>
+                {columns.map(({ label }, index) => (
+                  <Th
+                    key={index}
+                    sort={buildSort(index)}
+                  >
+                    {label}
+                  </Th>
```

Comment:
Looks like there is a way to give column widths: https://www.patternfly.org/v4/components/table#composable-cell-width-breakpoint-modifiers

Or the text in the column: https://www.patternfly.org/v4/components/table/#composable-controlling-text

I think it would be nicer to have the date column as no wrap.

Reply:
Done. Adding `breakWord` modifier on the message column prevents scrolling in case of long strings (i.e. names).